src/agent0/ethpy/base/transactions.py

```diff
@@ -433,22 +445,20 @@ def build_transaction(
 
     # Assign gas parameters
     # other than the optional gas parameter, this is the default behavior of web3py, exposed here for clarity
-    max_priority_fee = web3.eth.max_priority_fee
+    max_priority_fee = int(web3.eth.max_priority_fee * txn_options_priority_fee_multiple)
     pending_block = web3.eth.get_block("pending")
     base_fee = pending_block.get("baseFeePerGas", None)
+    if base_fee is not None:
```

Comment:
can't this be if/else?

Reply:
I think it would be identical, since they're float | None.
Seems more explicit though?
```
if txn_options_value is not None:
if base_fee is not None:
if txn_options_gas is not None:
```
vs.
```
if txn_options_value:
if base_fee:
if txn_options_gas:
```
seems like the latter forces you to think a bit more to understand what's going on, whereas the former makes it clear you're checking for non-default values.
would the latter fail if the value is 0? since it also checks for truthiness?

Reply:
sorry, I was talking  about

```py
if base_fee is not None:
  base_fee *= txn_options_base_fee_multiple
if base_fee is None:
  raise AssertionError("The latest block does not have a baseFeePerGas")
```

This should be `if ... else ...`

Regarding the other point you're making, I agree it's better to check the affirmative `is None` than the negative `is not None`. But `if base_fee` is **not** the same as `if base_fee is not None`. The latter would evaluate to `True` even if `base_fee = False`.

Reply:
changed in 48e0212a272a59992b47b0e5495acb93d498b820